game-core/src/main/java/games/strategy/triplea/delegate/RocketsFireHelper.java

```diff
@@ -68,64 +120,93 @@ private static boolean isLimitRocketDamageToProduction(final GameData data) {
     return Properties.getLimitRocketAndSbrDamageToProduction(data);
   }
 
-  static void fireRockets(final IDelegateBridge bridge, final PlayerID player) {
-    final GameData data = bridge.getData();
-    final Set<Territory> rocketTerritories = getTerritoriesWithRockets(data, player);
-    if (rocketTerritories.isEmpty()) {
-      bridge.getHistoryWriter().startEvent(player.getName() + " has no rockets to fire");
-      return;
-    }
-    if (isWW2V2(data) || isAllRocketsAttack(data)) {
-      fireWW2V2(bridge, player, rocketTerritories);
-    } else {
-      fireWW2V1(bridge, player, rocketTerritories);
-    }
-  }
-
-  private static void fireWW2V2(
-      final IDelegateBridge bridge,
-      final PlayerID player,
-      final Set<Territory> rocketTerritories) {
+  /**
+   *  Find Rocket Targets and load up fire rockets for later execution if necessary. Directly fired with strict rockets.
+   */
+  private void findRocketTargets(final IDelegateBridge bridge) {
```

Comment:
Isn't it actually firing the rockets if the new property is set at this line: https://github.com/triplea-game/triplea/pull/4324/files#diff-547357ca4f4a77daa0b2582028381c95R184

Reply:
You know what, you're quite correct. Even though I wrote all of that I had forgotten about that aspect.

I like the idea of breaking up the method so that it only selects a single rocket but that has two problems. 
(1) You need some kind of Tuple to return back both the territory and the unit - you could just continue with adding to the class members though. So not a show stopper by any means.
(2) You'd need to remember the previously attacked Territories. I guess you could make the previouslyAttackedTerritories variable a class member and remember it that way.

Do you want me to do that?

Reply:
I'd prefer to avoid class member variables generally as it tends to make things much harder to read and follow. Might be better to just have 2 different methods: 1 for finding all targets for non-global and 1 for dealing with target, fire, target, fire.

Reply:
Ultimately, I've gone with your original suggestion here of renaming the method.

A general comment on member variables. I've done a lot of stuff in a language that wouldn't allow you to omit the "this." on member variable access (although it called it "CurObject."). Python is similar. Have to say I much prefer this approach to the Java approach. However, given that the Java approach is to not distinguish between local variable access and member access, wouldn't it have been better to retain the "m_" prefix so that we could easily determine when this was done? Apologies if I've asked this before.

Is your issue with class member variables them not being obvious where they are coming from? In general I know what you mean about longer lived variables reducing readability but there's also issues if you pass around a tonne of variables too.